### PR TITLE
maintainers: litex: migrate from codeowners to maintainers.yml

### PR DESCRIPTION
**Pull request by maass-hamburg**
- add the `litex` prefix to the litex interrupt controller
- migrate the liteX codeowners to maintainers.yml
- add myself as collaborator to LiteX

### CODE REVIEW

File: dts/bindings/interrupt-controller/litex,vexriscv-intc0.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 description: LiteX VexRiscV interrupt controller
 
-compatible: "vexriscv-intc0"
+compatible: "litex,vexriscv-intc0"
```
Comment (MaureenHelm) [inline dts/bindings/interrupt-controller/litex,vexriscv-intc0.yaml:6]: Need to add a note to the 3.7 migration guide about the changed compatible

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/interrupt-controller/litex,vexriscv-intc0.yaml (hunk 1 above)
Comment (maass-hamburg) [reply dts/bindings/interrupt-controller/litex,vexriscv-intc0.yaml:6]: done

File: doc/releases/migration-guide-3.7.rst [hunk 2]
```diff
@@ -36,6 +36,9 @@ Boards
   ``SOC_IT81302CX``, ``SOC_IT82002AW``, ``SOC_IT82202AX``, ``SOC_IT82302AX``.
   And, rename the ``SOC_SERIES_ITE_IT8XXX2`` to ``SOC_SERIES_IT8XXX2``. (:github:`71680`)
 
+* LiteX: Renamed the ``compatible`` of the LiteX VexRiscV interrupt controller node from
+  :dtcompatible:`vexriscv-intc0` to :dtcompatible:`litex,vexriscv-intc0`. (:github:`73211`)
```
Comment (MaureenHelm) [inline doc/releases/migration-guide-3.7.rst:40]: I think referencing the old compatible with `:dtcompatible:` will lead to a broken link.

@kartben is that right?

Comment (maass-hamburg) [reply doc/releases/migration-guide-3.7.rst]: now changed

**Review by fkokosinski — COMMENTED** [review]
Hey, we are willing to keep maintaining this platform, so I've left a suggestion regarding its maintainership.

See: https://github.com/zephyrproject-rtos/zephyr/pull/73057

**Review by fkokosinski — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**